amical/_cli/commands/clean.py

```diff
@@ -0,0 +1,114 @@
+import os
+from datetime import datetime
+from glob import glob
+from pathlib import Path
+
+from astropy.io import fits
+from matplotlib import pyplot as plt
+from tabulate import tabulate
+from termcolor import cprint
+from tqdm import tqdm
+
+import amical
+
+
+def _select_data_file(args, process):
+    """Show report with the data found and allow to select one to be treated."""
+    l_file = sorted(glob("%s/*.fits" % args.datadir))
+
+    if len(l_file) == 0:
+        raise OSError("No fits files found in %s, check --datadir." % args.datadir)
+
+    headers = ["FILENAME", "TARGET", "DATE", "INSTRUM", "INDEX"]
+
+    index_file = []
+    d = []
+    for i, f in enumerate(l_file):
+        hdu = fits.open(f)
+        hdr = hdu[0].header
+        hdu.close()
+        target = hdr.get("OBJECT", None)
+        date = hdr.get("DATE-OBS", None)
+        ins = hdr.get("INSTRUME", None)
+        index_file.append(i)
+        filename = f.split("/")[-1]
+        d.append([filename, target, date, ins, i])
+
+    print(tabulate(d, headers=headers))
+
+    if args.file >= 0:
+        choosen_index = args.file
+    else:
+        choosen_index = int(input("\nWhich file to %s?\n" % process))
+
+    try:
+        filename = l_file[choosen_index]
+        with fits.open(filename) as hdul:
+            hdr = hdul[0].header
+    except IndexError:
+        raise IndexError(
+            "Selected index (%i) not valid (only %i files found)."
+            % (choosen_index, len(l_file))
+        )
+    return filename, hdr
+
+
+def perform_clean(args):
+    """CLI interface to clean the data with AMICAL."""
+    cprint("---- AMICAL clean process ----", "cyan")
+
+    clean_param = {
+        "isz": args.isz,
+        "r1": args.r1,
+        "dr": args.dr,
+        "apod": args.apod,
+        "window": args.window,
+        "f_kernel": args.kernel,
+    }
+
+    if not os.path.exists(args.datadir):
+        raise OSError(
+            "%s directory not found, check --datadir. AMICAL look for data only in this specified directory."
+            % args.datadir
+        )
```

Comment:
this is a CLI command, you shouldn't raise Python errors in there, but instead write an error message to `sys.stderr` and return an int other than 0